script.js

```diff
@@ -116,3 +139,70 @@ function sendMessage(groupId, message){
         error: function () { }
     });
 }
+
+
+function onConnectFailed(resp) {
+    console.log("onConnectFailed triggered");
+    if (navigator.onLine) {
+        console.log("reconnecting");
+        window.Applozic.ALSocket.reconnect();
+    }
+}
+function subscribeToTypingChannel(groupId){
+    window.Applozic.ALSocket.subscibeToTypingChannel(groupId);
+}
+
+function unsubscribeToTypingChannel(){
+    window.Applozic.ALSocket.unsubscibeToTypingChannel();
+}
+
+function subscribeToSupportChannel(){
+    window.Applozic.ALSocket.subscribeToSupportChannel();
+}
+function unsubscribeToSupportChannel(){
+    window.Applozic.ALSocket.unsubscribeToSupportChannel();
+}
+// anybody, (agent/user) when goes online, offline, or away, this will get triggered.
+//  KM_USER_STATUS: {
+    //     OFFLINE: 0,
+    //     ONLINE: 1,
+    //     AWAY: 2,
+    //     NOT_AWAY: 3,
+    // 
+// },
+function userStatusUpdateHandler(resp){
+    console.log('onUserStatusUpdate', JSON.parse(resp.body)["message"]);
```

Comment:
can you define variables and tell which part of msg is what. 
status = which key?
userName = which key?